src/parser2.rs

```diff
@@ -0,0 +1,312 @@
+            Token { rule: Rule::int, start: 0, end: 1 },
+            Token { rule: Rule::op_gt, start: 2, end: 3 },
+            Token { rule: Rule::int, start: 4, end: 5 },
+            Token { rule: Rule::op_or, start: 6, end: 8 },
+            Token { rule: Rule::and, start: 9, end: 25 },
+            Token { rule: Rule::comparison, start: 9, end: 15 },
+            Token { rule: Rule::int, start: 9, end: 10 },
+            Token { rule: Rule::op_eq, start: 11, end: 13 },
+            Token { rule: Rule::int, start: 14, end: 15 },
+            Token { rule: Rule::op_and, start: 16, end: 19 },
+            Token { rule: Rule::identifier, start: 20, end: 25 }
+        ]);
+    }
+
+    #[test]
+    fn test_if_tag() {
+        let mut parser = Rdp::new(StringInput::new("{% if true or show == false %}"));
+        assert!(parser.if_tag());
+        expect_tokens(&parser, vec![
+            Token { rule: Rule::if_tag, start: 0, end: 30 },
+            Token { rule: Rule::or, start: 6, end: 27 },
+            Token { rule: Rule::op_true, start: 6, end: 10 },
+            Token { rule: Rule::op_or, start: 11, end: 13 },
+            Token { rule: Rule::comparison, start: 14, end: 27 },
+            Token { rule: Rule::identifier, start: 14, end: 18 },
+            Token { rule: Rule::op_eq, start: 19, end: 21 },
+            Token { rule: Rule::op_false, start: 22, end: 27 }
+        ]);
+    }
+
+    #[test]
+    fn test_variable_tag() {
+        let mut parser = Rdp::new(StringInput::new("{{loop.index + 1}}"));
+        assert!(parser.variable_tag());
+        expect_tokens(&parser, vec![
+            Token { rule: Rule::variable_tag, start: 0, end: 18 },
+            Token { rule: Rule::add_sub, start: 2, end: 16 },
+            Token { rule: Rule::identifier, start: 2, end: 12 },
+            Token { rule: Rule::op_plus, start: 13, end: 14 },
+            Token { rule: Rule::int, start: 15, end: 16 }
+        ]);
+    }
+
+    #[test]
+    fn test_content() {
+        let mut parser = Rdp::new(StringInput::new("{% if i18n %}世界{% else %}world{% endif %}"));
+        assert!(parser.content());
+        expect_tokens(&parser, vec![
+            Token { rule: Rule::if_tag, start: 0, end: 13 },
+            Token { rule: Rule::identifier, start: 6, end: 10 },
+            Token { rule: Rule::text, start: 13, end: 19 },
+            Token { rule: Rule::else_tag, start: 19, end: 29 },
+            Token { rule: Rule::text, start: 29, end: 34 },
+            Token { rule: Rule::endif_tag, start: 34, end: 45 }
+        ]);
+    }
+
+    #[test]
+    fn test_template() {
+        let mut parser = Rdp::new(StringInput::new("
+            {# Greeter template #}
+            Hello {% if i18n %}世界{% else %}world{% endif %}
+            {% for country in countries %}
+                {{ loop.index }}.{{ country }}
+            {% endfor %}"));
+        assert!(parser.template());
+        expect_tokens(&parser, vec![
```

Comment:
Here is what I've changed:

```
text = { (!(block_start) ~ any )+ }
content = _{
    variable_tag |
    comment_tag |
    block_tag ~ content* ~ endblock_tag |
    if_tag ~ content* ~ (elif_tag ~ content?)* ~ (else_tag ~ content*)? ~ endif_tag |
    for_tag ~ content* ~ endfor_tag |
    text
}
```

I would also recommend using some kind of rule to enclose `content*` inside of, because matching `content*` inside the current `process!` macro might be hard.